homeassistant/components/qbittorrent/coordinator.py

```diff
@@ -39,20 +44,25 @@ def __init__(self, hass: HomeAssistant, client: Client) -> None:
             update_interval=timedelta(seconds=30),
         )
 
-    async def _async_update_data(self) -> dict[str, Any]:
-        """Async method to update QBittorrent data."""
+    async def _async_update_data(self) -> SyncMainDataDictionary:
         try:
-            return await self.hass.async_add_executor_job(self.client.sync_main_data)
-        except LoginRequired as exc:
-            raise HomeAssistantError(str(exc)) from exc
+            return await self.hass.async_add_executor_job(self.client.sync_maindata)
+        except (LoginFailed, Forbidden403Error) as exc:
+            raise HomeAssistantError(
+                translation_domain=DOMAIN, translation_key="login_error"
+            ) from exc
+        except APIConnectionError as exc:
+            raise HomeAssistantError(
+                translation_domain=DOMAIN, translation_key="cannot_connect"
+            ) from exc
 
-    async def get_torrents(self, torrent_filter: str) -> list[dict[str, Any]]:
+    async def get_torrents(self, torrent_filter: str) -> TorrentInfoList:
         """Async method to get QBittorrent torrents."""
         try:
             torrents = await self.hass.async_add_executor_job(
-                lambda: self.client.torrents(filter=torrent_filter)
+                lambda: self.client.torrents_info(torrent_filter)  # type: ignore[arg-type]
```

Comment:
Why is this and all other the ignore required?
Can we avoid them?

Reply:
The lib is using a weird typing, for example here, torrents_info is expecting `TorrentStatusesT` witch is a literal :
```python
#: Type for Torrent Status.
TorrentStatusesT = Literal[
    "all",
    "downloading",
    "seeding",
    "completed",
    ...
]
```
I think it's impossible to convert `str` to `Literal`, tell me if I'm wrong.

For those in `sensor.py`, it's because `SyncMainDataDictionary` is simply a `Dictionary[JsonValueT]` where `JsonValueT`: 

```python
JsonValueT = Union[
    None,
    int,
    str,
    bool,
    Sequence["JsonValueT"],
    Mapping[str, "JsonValueT"],
]
```
So the linter don't really like that.

If  you have another solution I'll be happy to change that.

I can maybe wrap all this in a ` if isinstanceof(var, int)` ?